beacon/engine/types.go

```diff
@@ -350,6 +351,48 @@ type ExecutionPayloadBody struct {
 	Withdrawals     []*types.Withdrawal `json:"withdrawals"`
 }
 
+// Max size of inclusion list in bytes.
+const MaxBytesPerInclusionList = uint64(8192)
+
+// InclusionList is a list of transactions that should be included in a payload.
+type InclusionList [][]byte
+
+func (inclusionList InclusionList) MarshalJSON() ([]byte, error) {
+	inclusionListHex := make([]hexutil.Bytes, len(inclusionList))
+	for i, tx := range inclusionList {
+		inclusionListHex[i] = tx
+	}
+
+	return json.Marshal(inclusionListHex)
+}
+
+func (inclusionList *InclusionList) UnmarshalJSON(input []byte) error {
+	var inclusionListHex []hexutil.Bytes
+	if err := json.Unmarshal(input, &inclusionListHex); err != nil {
+		return err
+	}
+
+	*inclusionList = make([][]byte, len(inclusionListHex))
+	for i, tx := range inclusionListHex {
+		(*inclusionList)[i] = tx
+	}
+
```

Comment:
```suggestion
	var il InclusionList
	for _, tx := range inclusionListHex {
		il = append(il, tx)
	}
	*inclusionList = il
```

Slightly cleaner version.

Reply:
The suggested version returns `nil` when inclusion list is empty. This cause `NewPayloadV5` to return invalid parameters error when it should accept an empty inclusion list. Thus, I'll stick to the previous version.